cmd/clusterctl/client/cluster/objectgraph.go

```diff
@@ -620,7 +622,18 @@ func (o *objectGraph) setSoftOwnership() {
 		for _, cluster := range clusters {
 			// if the cluster uses a managed topology and uses the clusterclass
 			// set the clusterclass as a soft owner of the cluster.
-			if className, ok := cluster.additionalInfo[clusterTopologyNameKey]; ok {
+			className, hasName := cluster.additionalInfo[clusterTopologyNameKey]
+			classNamespace, hasNamespace := cluster.additionalInfo[clusterTopologyNamespaceKey]
+			if hasNamespace && hasName {
+				if className == clusterClass.identity.Name && clusterClass.identity.Namespace == classNamespace {
+					cluster.addSoftOwner(clusterClass)
+
+					// Set isGlobalHierarchy on clusterClass if is referenced from a different namespace
+					if classNamespace != cluster.identity.Namespace {
+						clusterClass.isGlobalHierarchy = true
```

Comment:
While this probably achieve the goal, it seems a little bit hacky:
- isGlobal and isGlobalHierarchy are meant to track if "object is a global resource (no namespace).", which this is not the case for cc
- you are using isGlobalHierarchy without having an isGlobal object on top

Please take a look at the alternative solution I proposed

Reply:
Should be implemented “roughly” as described. I added a check to avoid deletion if multiple clusters reference the same `ClusterClass` (not sure if it was a part of original request but it seemed to make sense). May also work as single condition for delete across all namespaces too, but left it as requested.

Reply:
TBH I don't think that the number of cluster referencing a CC should be considered.
E.g. 
If I have a single ns with a CC, 10 cluster referencing the same CC (no cross namespace CC at all), it is perfectly ok to delete the CC after move.

It is also the current behaviour, so changing this might be considered a breaking change